mutable_tree.go

```diff
@@ -85,21 +85,23 @@ func (tree *MutableTree) Set(key, value []byte) bool {
 	return updated
 }
 
-func (tree *MutableTree) set(key []byte, value []byte) (orphaned []*Node, updated bool) {
+func (tree *MutableTree) set(key []byte, value []byte) (orphans []*Node, updated bool) {
 	if value == nil {
 		panic(fmt.Sprintf("Attempt to store nil value at key '%s'", key))
 	}
+
 	if tree.ImmutableTree.root == nil {
 		tree.ImmutableTree.root = NewNode(key, value, tree.version+1)
-		return nil, false
+		return nil, updated
 	}
-	tree.ImmutableTree.root, updated, orphaned = tree.recursiveSet(tree.ImmutableTree.root, key, value)
 
-	return orphaned, updated
+	orphans = make([]*Node, 0, tree.Height()+3)
```

Comment:
Again please add docstring explaining how we get this number.

`// Set will orphan at most tree.Height nodes, balancing the tree after a Set will orphan at most 3 nodes.`

Reply:
^As mentioned elsewhere, this should be abstracted into tiny helper function with documentation imo.

Reply:
sure